providers/providers.go

```diff
@@ -94,6 +99,16 @@ func newProviderDataFromConfig(providerConfig options.Provider) (*ProviderData,
 		p.Verifier = verifier
 		if oidcProvider != nil {
 			// Use the discovered values rather than any specified values
+			type DiscoveryClaims struct {
+				// RFC-8414 Authorization Server Metadata
+				CodeChallengeMethods []string `json:"code_challenge_methods_supported"`
+			}
```

Comment:
What happens for providers/users that aren't using discovery?

Reply:
I explained this in an earlier comment and in the commit message.
If PKCE is not detected as supported in the discovery document, it will be disabled to be safe. Users can enable it with the force CLI option.